test/e2e/subscription_e2e_test.go

```diff
@@ -1287,3 +1310,103 @@ func updateInternalCatalog(t *testing.T, c operatorclient.ClientInterface, crc v
 	})
 	require.NoError(t, err)
 }
+
+// TestSingleInstallPlanCreatedPerSubscription ensures that applying N number of subscriptions only creates
+// N number of InstallPlans.
+//
+// Steps:
+// 1. Create namespace, ns
+// 2. Create CatalogSource, cs, in ns
+// 3. Create multiple Subscriptions to each package of cs in ns, sub
+// 4. Wait for the package from sub to install successfully with no remaining InstallPlan status conditions
+// 5. Ensure that the InstallPlan is correctly referenced in the subscription
+// 6. Ensure that the number of InstallPlans created matches the number of subs created
+func TestSingleInstallPlanCreatedPerSubscription(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	c := newKubeClient(t)
+	crc := newCRClient(t)
+
+	// Create namespace ns
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: genName("ns-"),
+		},
+	}
+	_, err := c.KubernetesInterface().CoreV1().Namespaces().Create(ns)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, c.KubernetesInterface().CoreV1().Namespaces().Delete(ns.GetName(), &metav1.DeleteOptions{}))
+	}()
+
+	Numpkgs := 2
+
+	// Create a CatalogSource, cs, in ns with Numpkgs pkgs
+	pkgNames := []string{}
+	channelNames := []string{}
+	crds := []apiextensions.CustomResourceDefinition{}
+	csvs := []v1alpha1.ClusterServiceVersion{}
+	manifests := []registry.PackageManifest{}
+	for i := 0; i < Numpkgs; i++ {
+		pkgName := genName("pkg-")
+		channelName := genName("channel-")
+		strategy := newNginxInstallStrategy(pkgName, nil, nil)
+		crd := newCRD(pkgName)
+		csv := newCSV(pkgName, ns.GetName(), "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, strategy)
```

Comment:
this is putting n "0.1.0" releases in the configmap; it might be nice to increment it to simulate a real update chain.

Reply:
I'm not sure I understand why we want to simulate a real update chain - could you help clarify?

Reply:
This loop generates a set of csvs, crds, and packages, but they all have the same content. When we create the internal catalog source, they're all getting collapsed/ignored into 1 csv, 1 crd, and 1 package.

To rephrase my original comment - let's either remove the loop here (so there's really only 1 package/csv/crd), or make the loop generate content that can actually get loaded by a catalog by incrementing the versions, changing the names, setting up replacement chains, etc.

Unless I'm missing why this is the way it is?